src/main/java/io/github/soc/directories/Util.java

```diff
@@ -112,63 +137,49 @@ static String linuxExecutableDir(String homeDir, String dataDir) {
-    final String[] initialResult = getWinDirs(doubleEscapeQuotes, guids);
-
-    for (String s : initialResult) {
-      if (s != null)
-        return initialResult;
-    }
-
-    // First attempt failed, let's try to escape differently.
-    return getWinDirs(!doubleEscapeQuotes, guids);
-  }
-
-  static String[] getWinDirs(boolean doubleEscapeQuotes, String... guids) {
-
-    // Deal with legacy or safe handling of quotes by the JDK.
-    // Safe handling may be enabled for JDKs >= 1.8.0_231, under some conditions.
-    // See https://www.oracle.com/technetwork/java/javase/8u231-relnotes-5592812.html#JDK-8221858
-    // or https://github.com/AdoptOpenJDK/openjdk-jdk8u/commit/048eb42afa11ac217dcdb690d5b266fcb910771f
-    final String q = doubleEscapeQuotes ? "\\\"" : "\"";
-
     int guidsLength = guids.length;
     StringBuilder buf = new StringBuilder(guidsLength * 68);
     for (int i = 0; i < guidsLength; i++) {
-      buf.append("[Dir]::GetKnownFolderPath(" + q);
+      buf.append("[Dir]::GetKnownFolderPath(\"");
       buf.append(guids[i]);
-      buf.append(q + ")\n");
+      buf.append("\")\n");
     }
 
+    String encodedCommand = SCRIPT_START_BASE64 + toUTF16LEBase64(buf.toString() + "}");
+
     return runCommands(guidsLength, Charset.forName("UTF-8"),
         "powershell.exe",
         "-Command",
-        "& {\n" +
-            "[Console]::OutputEncoding = [System.Text.Encoding]::UTF8\n" +
-            "Add-Type @" + q + "\n" +
-            "using System;\n" +
-            "using System.Runtime.InteropServices;\n" +
-            "public class Dir {\n" +
-            "   [DllImport(" + q + "shell32.dll" + q + ")]\n" +
-            "   private static extern int SHGetKnownFolderPath([MarshalAs(UnmanagedType.LPStruct)] Guid rfid, uint dwFlags, IntPtr hToken, out IntPtr pszPath);\n" +
-            "   public static string GetKnownFolderPath(string rfid) {\n" +
-            "       IntPtr pszPath;\n" +
-            "       if (SHGetKnownFolderPath(new Guid(rfid), 0, IntPtr.Zero, out pszPath) != 0) return " + q + q + ";\n" +
-            "       string path = Marshal.PtrToStringUni(pszPath);\n" +
-            "       Marshal.FreeCoTaskMem(pszPath);\n" +
-            "       return path;\n" +
-            "   }\n" +
-            "}\n" +
-            q + "@\n" +
-            buf.toString() +
-            "}"
+        "-EncodedCommand",
+        encodedCommand
     );
   }
 
+  private static String toUTF16LEBase64(String script) {
+    byte[] scriptInUtf16LEBytes = script.getBytes(Charset.forName("UTF-16LE"));
+    if (base64EncodeMethod == null) {
+      initBase64Encoding();
+    }
+    try {
+      return (String) base64EncodeMethod.invoke(base64Encoder, scriptInUtf16LEBytes);
+    } catch (Exception e) {
+      throw new RuntimeException("Base64 encoding failed!", e);
+    }
+  }
+
+  private static void initBase64Encoding() {
+    try {
+      base64Encoder = Class.forName("java.util.Base64").getMethod("getEncoder").invoke(null);
```

Comment:
AFAIC, I'm fine with code requiring Java 8 now (`Base64`, `StandardCharsets`, etc.)

Reply:
I'm not. :-)

(More seriously: The idea is to keep this library Java 6 compatible forever, and publish the "new & shiny" version using modern features, modules, Project Panama under the new groupId `dev.dirs`.)

Reply:
Couldn't [multi-release JARs](http://openjdk.java.net/jeps/238) work here? via [sbt-multi-release-jar](https://github.com/sbt/sbt-multi-release-jar) or manually

Reply:
I'd rather release a completely legacy-free version when the JVM ships with support for it.
Currently I can also build everything in one go with an old JVM and have it work – this would at add at least 1 or 2 JDK variants I'd have to also keep working.